Production Fixes

brownd1978 opened the pull request:
Add ability to mix CosmicLivetime since that comes in through resampling.  Allow wildcarding process code in particle code filtering.  Fix IPA material name and parameters.

FNALbuild commented:
Hi @brownd1978,
You have proposed changes to files in these packages:
- Filters
- Analyses
- Mu2eG4
- EventMixing

which require these tests: build.

@Mu2e/fnalbuild-users, @Mu2e/write have access to CI actions on [master](/Mu2e/Offline/tree/master).

The following users requested to be notified about changes to these packages:
@resnegfk



:hourglass: The following tests have been triggered for a7b93dcdf680b7ea0af866a660654de968269707: build  (Build queue has 3 jobs)


[About FNALbuild.](https://mu2ewiki.fnal.gov/wiki/Git#GitHub_Pull_Request_Procedures_and_FNALbuild) [Code review on Mu2e/Offline.](https://mu2ewiki.fnal.gov/wiki/GitHubWorkflow#Code_Review)

gaponenko commented on Analyses/src/StoppedParticlesDumper_module.cc:
```diff
@@ -50,21 +50,26 @@ namespace mu2e {
       float tau; // proper time, for stopped pion weights
+      int pdg;
+      int origin;
+      int term;
 
-      StopInfo() : x(), y(), z(), t(), tau() {}
+      StopInfo() : x(), y(), z(), t(), tau(), pdg(0), origin(0), term(0) {}
```
I do not understand the purpose of the changes to StoppedParticlesDumper.  The sole purpose of the module was to provide inputs for stopped particle resampling; it is not a "general use" ntuple dumper.  With migration to the framework based resampling the logical thing would be to retire and delete this module, not to extend it.

brownd1978 replied:
This module is an analyzer, I extended its use as an analyzer, in order to study the output, which is what analyzers are supposed to do.  Its use in production is discontinued.  Perhaps the confusion is that there are still scripts that reference it in JobConfig, i will delete those.

gaponenko replied:
Thanks

gaponenko commented on EventMixing/inc/Mu2eProductMixer.hh:
```diff
@@ -140,14 +139,10 @@ namespace mu2e {
                           ExtMonFNALSimHitCollection& out,
                           art::PtrRemapper const& remap);
 
-    bool mixProtonBunchIntensity(std::vector<mu2e::ProtonBunchIntensity const*> const &in,
```
Why was mixProtonBunchIntensity() removed when adding mixCosmicLivetime()?

brownd1978 replied:
There is no use case for mixing this product.  Its only use case is in the main event stream of a mixing job.  I'm not sure of the history behind making it a potential mixin.

gaponenko commented on Filters/src/ParticleCodeFilter_module.cc:
```diff
@@ -32,9 +32,11 @@ namespace mu2e {
       struct ParticleCodeSelector {
 	PDGCode::type pdgCode_;
-	ProcessCode creationCode_, terminationCode_;
+	int creationCode_, terminationCode_;
```
PDGCode::type should be prefered to an int.    If a parameter is optional then std::optional<PDGCode::type> can be used here instead of relying on an expected range of "valid" integer values.

brownd1978 replied:
Formerly these variables were ProcessCode, PdgCode would not parse.  I will make them optional.

gaponenko commented on Mu2eG4/src/ConstructMaterials.cc:
```diff
@@ -259,8 +259,8 @@ namespace mu2e {
       G4Material* IPAPolyethylene = new G4Material( mat.name, 0.96*CLHEP::g/CLHEP::cm3, 2);
-      IPAPolyethylene->AddMaterial( findMaterialOrThrow("G4_H"), 0.14);
-      IPAPolyethylene->AddMaterial( findMaterialOrThrow("G4_C"), 0.86);
+      IPAPolyethylene->AddMaterial( findMaterialOrThrow("G4_H"), 0.12);
+      IPAPolyethylene->AddMaterial( findMaterialOrThrow("G4_C"), 0.88);
```
Polyethylene has 2:1 hydrogen:carbon molar ratios.  Then the old hydrogen mass fraction 0.14 is correct and the new one is wrong.   What am I missing?

brownd1978 replied:
According to Hank Glass the poly used for the IPA has ~2% carbon doping to make it conductive.  Note that the previous density was wrong.

gaponenko replied:
OK, this answers my question.

gaponenko reviewed (changes requested):
This looks like an assortment of unrelated changes some of which are straightforward and others I do not understand, see the comments below.

Also, I thought this was resolved in an earlier review, but I see that in ParticleCodeFilter_module.cc

ParticleCodeConfig codeConfig { fhicl::Name("ParticleCodes") };

still does not have a useful comment.   Can you please add one?
Andrei

brownd1978 commented:
Hi Andrei, yes this is a collection of small changes, I tried to list them in the PR description, sorry if that wasn't clear.  I will make the updates you request.
Dave

FNALbuild commented:
:question: The build job(s) have failed or timed out on Jenkins, as there has been no result for over an hour.


- build ([more info](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//738/console))

The tests may now be triggered again.

gaponenko commented:
On Mon, 21 Jun 2021, David Brown wrote:

> @brownd1978 commented on this pull request.
> 
> 
> 
> > @@ -140,14 +139,10 @@ namespace mu2e {
>                            ExtMonFNALSimHitCollection& out,
>                            art::PtrRemapper const& remap);
>  
> -    bool mixProtonBunchIntensity(std::vector<mu2e::ProtonBunchIntensity const*> const &in,
> 
> There is no use case for mixing this product.  Its only use case is in the main event stream of a mixing job.  I'm not sure of the history behind making it a potential mixin.

But then there is no use case for mixing CosmicLiveTime (and the code says 
it can not be mixed).  Why did we add cosmic mixing then?

Andrei

brownd1978 commented:
David Brown

> On Jun 21, 2021, at 18:12, Andrei Gaponenko ***@***.***> wrote:
> 
> ﻿
> On Mon, 21 Jun 2021, David Brown wrote:
> 
> > @brownd1978 commented on this pull request.
> > 
> > 
> > 
> > > @@ -140,14 +139,10 @@ namespace mu2e {
> > ExtMonFNALSimHitCollection& out,
> > art::PtrRemapper const& remap);
> > 
> > - bool mixProtonBunchIntensity(std::vector<mu2e::ProtonBunchIntensity const*> const &in,
> > 
> > There is no use case for mixing this product. Its only use case is in the main event stream of a mixing job. I'm not sure of the history behind making it a potential mixin.
> 
> But then there is no use case for mixing CosmicLiveTime (and the code says 
> it can not be mixed). Why did we add cosmic mixing then?

It is resampled, and that uses the same product mixer as mixing.  Its needed to normalize the cosmic outputs. 

> 
> Andrei
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

gaponenko commented:
>> Why did we add cosmic mixing then?
> It is resampled, and that uses the same product mixer as mixing. Its needed to normalize the cosmic outputs.

I see.  It is a record of generator settings in the first simulation stage.  So this is a cosmic-specific piece
of information that should be used together with GenEventCount to do the normalization.  Is that right?
Andrei

kutschke commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for a7b93dcdf680b7ea0af866a660654de968269707: build  (Build queue has 1 jobs)

brownd1978 commented:
correct

On Mon, Jun 21, 2021 at 9:11 PM Andrei Gaponenko ***@***.***>
wrote:

> Why did we add cosmic mixing then?
> It is resampled, and that uses the same product mixer as mixing. Its
> needed to normalize the cosmic outputs.
>
> I see. It is a record of generator settings in the first simulation stage.
> So this is a cosmic-specific piece
> of information that should be used together with GenEventCount to do the
> normalization. Is that right?
> Andrei
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#issuecomment-865513049>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH574ZMMI5YQINXKB6Z4LTUAEN3ANCNFSM47CRZ6HQ>
> .
>


-- 
David Nathan Brown            ***@***.***
Office Phone (510) 486-7261  Fax  495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C)  Berkeley, CA 94720

FNALbuild commented:
:question: The build job(s) have failed or timed out on Jenkins, as there has been no result for over an hour.


- build ([more info](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//739/console))

The tests may now be triggered again.

brownd1978 commented:
I've addressed the reviewer comments.  I couldn't find a way to provide default arguments or make elements optional for a fhicl::Tuple, so that request wasn't implemented.

kutschke commented:
Andrei: if you are satisfied with Dave's replies, please approve the PR

gaponenko reviewed (changes requested):
Hi Dave,

The adding comments in the material definition is a great idea; otherwise some one could fix the "wrong" fractions in the future.

In the selector code, ProcessCode::unknown  is an alternative to using std::optional, fine with me.
The codeConfig comment says "PDG, and optionally creation or termination code".
However if one does not provide either of the codes the job would crash.
I think all 3 values must always be provided.    Should the comment say something like
"PDG, creation code, termination code.  Use process code=0 to not cut on that value."

Andrei

kutschke commented:
I merged in Ryun's updates to codetools that will catch stuck tests and report that they are stuck.
Retesting this PR now.

kutschke commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for b97b7dc4813f6b786486b00aa7241949bf566a2f: build  (Build queue has 3 jobs)

FNALbuild commented:
:sunny: The tests passed at b97b7dc4813f6b786486b00aa7241949bf566a2f.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged b97b7dc4813f6b786486b00aa7241949bf566a2f at 3482160ce8f9f05b46a5ccfc9560c167ec3f6327 |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/scons.log). Build time: 13 min 50 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/cosmicSimReco.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/g4surfaceCheck.log)  |
| FIXME, TODO count | :wavy_dash: | [TODO (0) FIXME (6) in 5 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//740/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
Andrei: has Dave addressed your last concern?  Today I am leaving early, about 4:15 PM.

gaponenko reviewed (changes requested):
Hi Dave,

If I select 'unknown' I get the error message below.

Andrei

$ mu2e -c Production/JobConfig/beam/MuonStopSelector.fcl
%MSG-e duplicateDestination: 

============================================================================

    Duplicate name for a MessageLogger Statistics destination: "cout"
    Only original configuration instructions are used. 

============================================================================

%MSG
%MSG-s ArtException:  ParticleCodeFilter:muminusSelector@Construction  22-Jun-2021 15:10:08 CDT ModuleConstruction
cet::exception caught in art
---- Type mismatch BEGIN
  
  Unsuccessful attempt to convert FHiCL parameter 'ParticleCodes[0][1]' to type 'int'.
  
  [Specific error:]
  ---- Type mismatch BEGIN
    error in signed string:
    unknown
    at or before:
  ---- Type mismatch END
---- Type mismatch END
%MSG

kutschke commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for 228d547603153bc8cb64db01b2f97001dd5074f4: build  (Build queue has 2 jobs)

gaponenko commented:
ProcessCode can be set from a string, why not actually use that instead of 
numbers?  PDGId can stay as a number (tuples can be heterogeneous); I do 
not see a string-to-PDGId support readily available.

Andrei

On Tue, 22 Jun 2021, Andrei Gaponenko wrote:

> @gaponenko requested changes on this pull request.
> 
> Hi Dave,
> 
> If I select 'unknown' I get the error message below.
> 
> Andrei
> 
> $ mu2e -c Production/JobConfig/beam/MuonStopSelector.fcl
> %MSG-e duplicateDestination: 
> 
> ============================================================================
> 
>     Duplicate name for a MessageLogger Statistics destination: "cout"
>     Only original configuration instructions are used. 
> 
> ============================================================================
> 
> %MSG
> %MSG-s ArtException:  ParticleCodeFilter:muminusSelector@Construction  22-Jun-2021 15:10:08 CDT ModuleConstruction
> cet::exception caught in art
> ---- Type mismatch BEGIN
>   
>   Unsuccessful attempt to convert FHiCL parameter 'ParticleCodes[0][1]' to type 'int'.
>   
>   [Specific error:]
>   ---- Type mismatch BEGIN
>     error in signed string:
>     unknown
>     at or before:
>   ---- Type mismatch END
> ---- Type mismatch END
> %MSG
> 
> 
> 
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23pullrequestreview-2D689952087&d=DwICaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=1I6_etjhHdSt43g8M6inY9gJUgF5ax2RTM-Zk7Lgn-c&s=XXYgjf6rdF3PmzqcohfqnH0huMvUt1MlMjNNaM79kpI&e=

brownd1978 commented:
you need the head of Production, since the type of the argument changed.
Dave

On Tue, Jun 22, 2021 at 1:11 PM Andrei Gaponenko ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi Dave,
>
> If I select 'unknown' I get the error message below.
>
> Andrei
>
> $ mu2e -c Production/JobConfig/beam/MuonStopSelector.fcl
> %MSG-e duplicateDestination:
>
>
> ============================================================================
>
> Duplicate name for a MessageLogger Statistics destination: "cout"
> Only original configuration instructions are used.
>
>
> ============================================================================
>
> %MSG
> %MSG-s ArtException: ParticleCodeFilter:muminusSelector@Construction
> 22-Jun-2021 15:10:08 CDT ModuleConstruction
> cet::exception caught in art
> ---- Type mismatch BEGIN
>
> Unsuccessful attempt to convert FHiCL parameter 'ParticleCodes[0][1]' to
> type 'int'.
>
> [Specific error:]
> ---- Type mismatch BEGIN
> error in signed string:
> unknown
> at or before:
> ---- Type mismatch END
> ---- Type mismatch END
> %MSG
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#pullrequestreview-689952087>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH5757FHFE7NLDC7Q5HZTTUDU6VANCNFSM47CRZ6HQ>
> .
>


-- 
David Nathan Brown            ***@***.***
Office Phone (510) 486-7261  Fax  495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C)  Berkeley, CA 94720

brownd1978 commented:
There is a mismatch that some of our enums are classes others just
namespaces.  My intent is to get production running, not to try to rectify
every problem from the past.
I think this is good enough.

On Tue, Jun 22, 2021 at 1:23 PM Andrei Gaponenko ***@***.***>
wrote:

> ProcessCode can be set from a string, why not actually use that instead of
> numbers? PDGId can stay as a number (tuples can be heterogeneous); I do
> not see a string-to-PDGId support readily available.
>
> Andrei
>
> On Tue, 22 Jun 2021, Andrei Gaponenko wrote:
>
> > @gaponenko requested changes on this pull request.
> >
> > Hi Dave,
> >
> > If I select 'unknown' I get the error message below.
> >
> > Andrei
> >
> > $ mu2e -c Production/JobConfig/beam/MuonStopSelector.fcl
> > %MSG-e duplicateDestination:
> >
> >
> ============================================================================
> >
> > Duplicate name for a MessageLogger Statistics destination: "cout"
> > Only original configuration instructions are used.
> >
> >
> ============================================================================
> >
> > %MSG
> > %MSG-s ArtException: ParticleCodeFilter:muminusSelector@Construction
> 22-Jun-2021 15:10:08 CDT ModuleConstruction
> > cet::exception caught in art
> > ---- Type mismatch BEGIN
> >
> > Unsuccessful attempt to convert FHiCL parameter 'ParticleCodes[0][1]' to
> type 'int'.
> >
> > [Specific error:]
> > ---- Type mismatch BEGIN
> > error in signed string:
> > unknown
> > at or before:
> > ---- Type mismatch END
> > ---- Type mismatch END
> > %MSG
> >
> >
> >
> >
> > --
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly or view it on GitHub:
> >
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23pullrequestreview-2D689952087&d=DwICaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=1I6_etjhHdSt43g8M6inY9gJUgF5ax2RTM-Zk7Lgn-c&s=XXYgjf6rdF3PmzqcohfqnH0huMvUt1MlMjNNaM79kpI&e=
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#issuecomment-866307308>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH5745PPVATKMTEB5YXMDTUDWKNANCNFSM47CRZ6HQ>
> .
>


-- 
David Nathan Brown            ***@***.***
Office Phone (510) 486-7261  Fax  495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C)  Berkeley, CA 94720

gaponenko commented:
I got the head of Productions and updated Offline to ipafix from your 
repository, then edited the fcl file to use 'unknown' per

  mu2e --print-description ParticleCodeFilter

I do not see how this could possibly work, the config type
is declared as  fhicl::Tuple<int,int,int>
so a non-numeric input like 'unknown' would not parse.
It would make sense to switch to 
fhicl::Tuple<int,string,string>, see my other comment.

Andrei

On Tue, 22 Jun 2021, David Brown wrote:

> you need the head of Production, since the type of the argument changed.
> Dave
> 
> On Tue, Jun 22, 2021 at 1:11 PM Andrei Gaponenko ***@***.***>
> wrote:
> 
> > ***@***.**** requested changes on this pull request.
> >
> > Hi Dave,
> >
> > If I select 'unknown' I get the error message below.
> >
> > Andrei
> >
> > $ mu2e -c Production/JobConfig/beam/MuonStopSelector.fcl
> > %MSG-e duplicateDestination:
> >
> >
> > ============================================================================
> >
> > Duplicate name for a MessageLogger Statistics destination: "cout"
> > Only original configuration instructions are used.
> >
> >
> > ============================================================================
> >
> > %MSG
> > %MSG-s ArtException: ParticleCodeFilter:muminusSelector@Construction
> > 22-Jun-2021 15:10:08 CDT ModuleConstruction
> > cet::exception caught in art
> > ---- Type mismatch BEGIN
> >
> > Unsuccessful attempt to convert FHiCL parameter 'ParticleCodes[0][1]' to
> > type 'int'.
> >
> > [Specific error:]
> > ---- Type mismatch BEGIN
> > error in signed string:
> > unknown
> > at or before:
> > ---- Type mismatch END
> > ---- Type mismatch END
> > %MSG
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23pullrequestreview-2D689952087&d=DwIFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=7J8nxEzAkrhU_xvYEDk_o0Z9-PJJX2ydS1jLs0qJvoE&s=mii3pwCIwDjfEdS8r6N7y78oYWBwKnTK9_Wze6EI9T8&e= >,
> > or unsubscribe
> > <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABAH5757FHFE7NLDC7Q5HZTTUDU6VANCNFSM47CRZ6HQ&d=DwIFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=7J8nxEzAkrhU_xvYEDk_o0Z9-PJJX2ydS1jLs0qJvoE&s=OXdvJ4PD_pHe3wD-Sb8TaPAeSj319sYUaBoX5ySSuTU&e= >
> > .
> >
> 
> 
> -- 
> David Nathan Brown            ***@***.***
> Office Phone (510) 486-7261  Fax  495-2957
> Lawrence Berkeley National Lab
> MS 50R5008 (50-6026C)  Berkeley, CA 94720
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866307690&d=DwIFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=7J8nxEzAkrhU_xvYEDk_o0Z9-PJJX2ydS1jLs0qJvoE&s=7eJahWeSfedjK3hLOelAvAoCxY10eSV6YhYbShjhR50&e=

gaponenko commented:
We will never have time and resourced to do a dedicated "clean up" of
Offline code.  It makes sense to use the pull request review process
to at least avoid adding more "to be fixed later" code that can be
easily improved on the spot instead.  It's not like we will miss a DOE
review deadline if MCD2020 is run a bit earlier or later.

Andrei

On Tue, 22 Jun 2021, David Brown wrote:

> There is a mismatch that some of our enums are classes others just
> namespaces.  My intent is to get production running, not to try to rectify
> every problem from the past.
> I think this is good enough.
> 
> On Tue, Jun 22, 2021 at 1:23 PM Andrei Gaponenko ***@***.***>
> wrote:
> 
> > ProcessCode can be set from a string, why not actually use that instead of
> > numbers? PDGId can stay as a number (tuples can be heterogeneous); I do
> > not see a string-to-PDGId support readily available.
> >
> > Andrei
> >
> > On Tue, 22 Jun 2021, Andrei Gaponenko wrote:
> >
> > > @gaponenko requested changes on this pull request.
> > >
> > > Hi Dave,
> > >
> > > If I select 'unknown' I get the error message below.
> > >
> > > Andrei
> > >
> > > $ mu2e -c Production/JobConfig/beam/MuonStopSelector.fcl
> > > %MSG-e duplicateDestination:
> > >
> > >
> > ============================================================================
> > >
> > > Duplicate name for a MessageLogger Statistics destination: "cout"
> > > Only original configuration instructions are used.
> > >
> > >
> > ============================================================================
> > >
> > > %MSG
> > > %MSG-s ArtException: ParticleCodeFilter:muminusSelector@Construction
> > 22-Jun-2021 15:10:08 CDT ModuleConstruction
> > > cet::exception caught in art
> > > ---- Type mismatch BEGIN
> > >
> > > Unsuccessful attempt to convert FHiCL parameter 'ParticleCodes[0][1]' to
> > type 'int'.
> > >
> > > [Specific error:]
> > > ---- Type mismatch BEGIN
> > > error in signed string:
> > > unknown
> > > at or before:
> > > ---- Type mismatch END
> > > ---- Type mismatch END
> > > %MSG
> > >
> > >
> > >
> > >
> > > --
> > > You are receiving this because you are subscribed to this thread.
> > > Reply to this email directly or view it on GitHub:
> > >
> > https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23pullrequestreview-2D689952087&d=DwICaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=1I6_etjhHdSt43g8M6inY9gJUgF5ax2RTM-Zk7Lgn-c&s=XXYgjf6rdF3PmzqcohfqnH0huMvUt1MlMjNNaM79kpI&e=
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866307308&d=DwIFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=C1vFA1JGVMUao8Qk4zS6h2HzXAXXGMkn_lSi_OVn-6A&s=iBclZaMnWmDyhH4vzVY7ZZF-1fdFkXz7iOT87udhhMI&e= >, or
> > unsubscribe
> > <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABAH5745PPVATKMTEB5YXMDTUDWKNANCNFSM47CRZ6HQ&d=DwIFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=C1vFA1JGVMUao8Qk4zS6h2HzXAXXGMkn_lSi_OVn-6A&s=77BFZKT2bZ5oUBSAyp2MXzXtGplTITS9gAHVK3GJ70w&e= >
> > .
> >
> 
> 
> -- 
> David Nathan Brown            ***@***.***
> Office Phone (510) 486-7261  Fax  495-2957
> Lawrence Berkeley National Lab
> MS 50R5008 (50-6026C)  Berkeley, CA 94720
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866312472&d=DwIFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O47fc5vzDTR2V_gla4Ub0Q&m=C1vFA1JGVMUao8Qk4zS6h2HzXAXXGMkn_lSi_OVn-6A&s=_xRBtnAoCwbJoC_xFfXjaCH4Mi0Bq8sh5JxQuqpe044&e=

kutschke commented:
Hi Andrei,

  In principle you are right.  But as a practical matter I want to move ahead now.

  I am leaving for the day at about 4:20 PM today and would appreciate if  you approve it before then so that I can merge before I leave.

  Thanks,
                  Rob


From: Andrei Gaponenko ***@***.***>
Reply-To: Mu2e/Offline ***@***.***>
Date: Tuesday, June 22, 2021 at 3:39 PM
To: Mu2e/Offline ***@***.***>
Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
Subject: Re: [Mu2e/Offline] Production Fixes (#501)

We will never have time and resourced to do a dedicated "clean up" of
Offline code. It makes sense to use the pull request review process
to at least avoid adding more "to be fixed later" code that can be
easily improved on the spot instead. It's not like we will miss a DOE
review deadline if MCD2020 is run a bit earlier or later.

Andrei

On Tue, 22 Jun 2021, David Brown wrote:

> There is a mismatch that some of our enums are classes others just
> namespaces. My intent is to get production running, not to try to rectify
> every problem from the past.
> I think this is good enough.

FNALbuild commented:
:umbrella: The tests failed for 228d547603153bc8cb64db01b2f97001dd5074f4.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged 228d547603153bc8cb64db01b2f97001dd5074f4 at 3482160ce8f9f05b46a5ccfc9560c167ec3f6327 |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/scons.log). Build time: 14 min 01 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/cosmicSimReco.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/ceSteps.log)  |
| ceDigi | :stopwatch: :x: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/ceDigi.log) Timed out. |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/g4surfaceCheck.log)  |
| FIXME, TODO count | :wavy_dash: | [TODO (0) FIXME (6) in 5 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//741/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

ryuwd commented:
FYI a test job is now monitored closely for time and killed if it doesn't finish 20 minutes after being started. 

This lets FNALbuild post the usual detailed summary (including showing the tests that timed out) rather than the cryptic "the job stalled" notification which is caused by the hard timeout on the jenkins job stopping everything after 1h30m.

brownd1978 commented:
OK I changed to string.

brownd1978 commented:
@FNALbuild run build tests

FNALbuild commented:
:hourglass: The following tests have been triggered for e7882297d0d23534dbcbe428245a991ef49fad74: build  (Build queue has 3 jobs)

FNALbuild commented:
:umbrella: The tests failed for e7882297d0d23534dbcbe428245a991ef49fad74.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| merge | :white_check_mark: | Merged e7882297d0d23534dbcbe428245a991ef49fad74 at 3482160ce8f9f05b46a5ccfc9560c167ec3f6327 |
| scons build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/scons.log). Build time: 12 min 43 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/cosmicSimReco.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/ceSteps.log)  |
| ceDigi | :stopwatch: :x: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/ceDigi.log) Timed out. |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/g4surfaceCheck.log)  |
| FIXME, TODO count | :wavy_dash: | [TODO (0) FIXME (6) in 5 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/fixme_todo.log) |
| clang-tidy | :wavy_dash: | [0 errors 0 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/clang-tidy.log) |


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

brownd1978 commented:
I think this is a real failure, not a timeout, as it always stops at the
same point, and none of the other (longer) jobs show problems.   If I run
the job sequence ceSteps -> ceDigi interactively there is no problem, so it
must be due to some difference in the CI environment vs the interactive
environment.  I'm unsure how to progress debugging this.

On Tue, Jun 22, 2021 at 3:27 PM Fermilab build bot account <
***@***.***> wrote:

> ☔ The tests failed for e788229
> <https://github.com/Mu2e/Offline/commit/e7882297d0d23534dbcbe428245a991ef49fad74>
> .
> Test Result Details
> merge ✅ Merged e788229
> <https://github.com/Mu2e/Offline/commit/e7882297d0d23534dbcbe428245a991ef49fad74>
> at 3482160
> <https://github.com/Mu2e/Offline/commit/3482160ce8f9f05b46a5ccfc9560c167ec3f6327>
> scons build (prof) ✅ Log file
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/scons.log>.
> Build time: 12 min 43 sec
> ceSimReco ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/ceSimReco.log>
> g4test_03MT ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/g4test_03MT.log>
> transportOnly ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/transportOnly.log>
> POT ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/POT.log>
> g4study ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/g4study.log>
> cosmicSimReco ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/cosmicSimReco.log>
> ceSteps ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/ceSteps.log>
> ceDigi ⏱️ ❌ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/ceDigi.log>
> Timed out.
> muDauSteps ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/muDauSteps.log>
> ceMix ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/ceMix.log>
> rootOverlaps ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/rootOverlaps.log>
> g4surfaceCheck ✅ Log file.
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/g4surfaceCheck.log>
> FIXME, TODO count 〰️ TODO (0) FIXME (6) in 5 files
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/fixme_todo.log>
> clang-tidy 〰️ 0 errors 0 warnings
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/artifact/clang-tidy.log>
>
> For more information, please check the job page here
> <https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//742/console>
> .
> Build artifacts are deleted after 5 days. If this is not desired, select Keep
> this build forever on the job page.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#issuecomment-866378775>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH574XGCGUVMBWSOYAK7LTUEE67ANCNFSM47CRZ6HQ>
> .
>


-- 
David Nathan Brown            ***@***.***
Office Phone (510) 486-7261  Fax  495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C)  Berkeley, CA 94720

ryuwd commented:
Looking at https://github.com/brownd1978/Offline/commits/ipafix I notice this branch was created off  4ae3e29fcebcc9f5853b9541acc4c3428f0a0cf1 which is before 3482160ce8f9f05b46a5ccfc9560c167ec3f6327 (where the ceDigi failures started)

FNALbuild does a merge into the latest commit on master first before building the result of that and running the tests. So if #491 introduced the problem then your interactive test on this branch would not be able to reproduce the CI result.

Hope that helps

ryuwd commented:
One more thing,

This should reproduce the state of the repository that the CI uses
```
cd Offline  # (presumably clone of fork)
git remote add upstream https://github.com/Mu2e/Offline
git fetch upstream
git checkout -b ipafix-merged-result ipafix
git merge --no-ff upstream/master -m "merged PR result"

# build
# run tests
```

gaponenko reviewed (approved):
Very nice!  Thank you, Dave!
Andrei

kutschke commented:
Dave: when you say that you tested interactively, do you mean on your branch or did you do a trial merge with master?

kutschke commented:
Is there still an overall timeout or is there still only a series of individual timeouts?

From: ryuwd ***@***.***>
Reply-To: Mu2e/Offline ***@***.***>
Date: Tuesday, June 22, 2021 at 4:25 PM
To: Mu2e/Offline ***@***.***>
Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
Subject: Re: [Mu2e/Offline] Production Fixes (#501)


FYI a test job is now monitored closely for time and killed if it doesn't finish 20 minutes after being started.

This lets FNALbuild post the usual detailed summary (including showing the tests that timed out) rather than the cryptic "the job stalled" notification which is caused by the hard timeout on the jenkins job stopping everything after 1h30m.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866344994&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Fn5oYqf98_rECKP0V6ZC9oEW08iN2-vHyUU9SObbPHo&s=qlehTdggVaLrxw_62oGqGvM0UDgB3tt-XLYDPZkppKc&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZSRACKRX5AHOVZHJQLTUD5UDANCNFSM47CRZ6HQ&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Fn5oYqf98_rECKP0V6ZC9oEW08iN2-vHyUU9SObbPHo&s=OWDOBijLuvmJm4j9iE1Bu4p1NFf7iLdLFedTsWRA8B4&e=>.

brownd1978 commented:
I tested interactively on the same branch that has stalled in CI.

On Tue, Jun 22, 2021 at 7:04 PM Rob Kutschke ***@***.***>
wrote:

> Dave: when you say that you tested interactively, do you mean on your
> branch or did you do a trial merge with master?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#issuecomment-866463611>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH57ZM7Y4QCM7OWZ7QIWLTUE6JHANCNFSM47CRZ6HQ>
> .
>


-- 
David Nathan Brown            ***@***.***
Office Phone (510) 486-7261  Fax  495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C)  Berkeley, CA 94720

kutschke commented:
HI Dave,

   The branch that stalls in CI is your branch merged into the current head of master.  Is that what you mean?

Rob

From: David Brown ***@***.***>
Reply-To: Mu2e/Offline ***@***.***>
Date: Tuesday, June 22, 2021 at 9:07 PM
To: Mu2e/Offline ***@***.***>
Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
Subject: Re: [Mu2e/Offline] Production Fixes (#501)

I tested interactively on the same branch that has stalled in CI.

On Tue, Jun 22, 2021 at 7:04 PM Rob Kutschke ***@***.***>
wrote:

> Dave: when you say that you tested interactively, do you mean on your
> branch or did you do a trial merge with master?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#issuecomment-866463611><https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866463611-253E&d=DwQFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=F4SAQXZfFnmEU9GhkC4oq1vIQl9i9ex-LbOeJyrxL8g&s=eHyhnlN13DEEE-Cr3McnlVJH5OnC1IXDvPaM6gpivp0&e=>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH57ZM7Y4QCM7OWZ7QIWLTUE6JHANCNFSM47CRZ6HQ><https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABAH57ZM7Y4QCM7OWZ7QIWLTUE6JHANCNFSM47CRZ6HQ-253E&d=DwQFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=F4SAQXZfFnmEU9GhkC4oq1vIQl9i9ex-LbOeJyrxL8g&s=45Q3EDgSbXgV-5iUzVJ0nsgO3hI_-ZrWv6zW9b3ywAY&e=>
> .
>


--
David Nathan Brown ***@***.***
Office Phone (510) 486-7261 Fax 495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C) Berkeley, CA 94720


—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866465266&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=F4SAQXZfFnmEU9GhkC4oq1vIQl9i9ex-LbOeJyrxL8g&s=Eesdn39gbDxrs7xVKFs509Pl5y28rMARZbg_B3iKAS8&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZQ5LWCM5EPZS73HDJLTUE6XFANCNFSM47CRZ6HQ&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=F4SAQXZfFnmEU9GhkC4oq1vIQl9i9ex-LbOeJyrxL8g&s=PA6cu67lAwSSHWwSmwT8rvtWH2mzUCt5gruNaAuygkY&e=>.

kutschke commented:
FYI.  I have  a test merge of your branch into master and am building now.

From: David Brown ***@***.***>
Reply-To: Mu2e/Offline ***@***.***>
Date: Tuesday, June 22, 2021 at 9:07 PM
To: Mu2e/Offline ***@***.***>
Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
Subject: Re: [Mu2e/Offline] Production Fixes (#501)

I tested interactively on the same branch that has stalled in CI.

On Tue, Jun 22, 2021 at 7:04 PM Rob Kutschke ***@***.***>
wrote:

> Dave: when you say that you tested interactively, do you mean on your
> branch or did you do a trial merge with master?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#issuecomment-866463611><https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866463611-253E&d=DwQFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=F4SAQXZfFnmEU9GhkC4oq1vIQl9i9ex-LbOeJyrxL8g&s=eHyhnlN13DEEE-Cr3McnlVJH5OnC1IXDvPaM6gpivp0&e=>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH57ZM7Y4QCM7OWZ7QIWLTUE6JHANCNFSM47CRZ6HQ><https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABAH57ZM7Y4QCM7OWZ7QIWLTUE6JHANCNFSM47CRZ6HQ-253E&d=DwQFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=F4SAQXZfFnmEU9GhkC4oq1vIQl9i9ex-LbOeJyrxL8g&s=45Q3EDgSbXgV-5iUzVJ0nsgO3hI_-ZrWv6zW9b3ywAY&e=>
> .
>


--
David Nathan Brown ***@***.***
Office Phone (510) 486-7261 Fax 495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C) Berkeley, CA 94720


—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866465266&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=F4SAQXZfFnmEU9GhkC4oq1vIQl9i9ex-LbOeJyrxL8g&s=Eesdn39gbDxrs7xVKFs509Pl5y28rMARZbg_B3iKAS8&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZQ5LWCM5EPZS73HDJLTUE6XFANCNFSM47CRZ6HQ&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=F4SAQXZfFnmEU9GhkC4oq1vIQl9i9ex-LbOeJyrxL8g&s=PA6cu67lAwSSHWwSmwT8rvtWH2mzUCt5gruNaAuygkY&e=>.

kutschke commented:
I did the merge and build.
I ran 100 events of ceSteps; it writes 58 events.

Then I ran   time mu2e -c Validation/ceDigi.fcl --trace

This sometimes runs to completion and sometimes hangs in makeSD.

It hangs maybe 1 time in 3 or 4.  When it’s hanging it’s consuming cpu.  So I presume infinite loop not IO wait.

Looking to see if it stops at the same event or different ones/



@reply.github.com>
Date: Tuesday, June 22, 2021 at 9:07 PM
To: Mu2e/Offline ***@***.***>
Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
Subject: Re: [Mu2e/Offline] Production Fixes (#501)

I tested interactively on the same branch that has stalled in CI.

brownd1978 commented:
Could you attach gdb when it hangs and find out what function it's in?

On Tue, Jun 22, 2021 at 7:37 PM Rob Kutschke ***@***.***>
wrote:

> I did the merge and build.
> I ran 100 events of ceSteps; it writes 58 events.
>
> Then I ran time mu2e -c Validation/ceDigi.fcl --trace
>
> This sometimes runs to completion and sometimes hangs in makeSD.
>
> It hangs maybe 1 time in 3 or 4. When it’s hanging it’s consuming cpu. So
> I presume infinite loop not IO wait.
>
> Looking to see if it stops at the same event or different ones/
>
>
>
> @reply.github.com>
> Date: Tuesday, June 22, 2021 at 9:07 PM
> To: Mu2e/Offline ***@***.***>
> Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
> Subject: Re: [Mu2e/Offline] Production Fixes (#501)
>
> I tested interactively on the same branch that has stalled in CI.
>
>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#issuecomment-866476011>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH57YZHTWEYTUKV5EOO6TTUFCH7ANCNFSM47CRZ6HQ>
> .
>


-- 
David Nathan Brown            ***@***.***
Office Phone (510) 486-7261  Fax  495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C)  Berkeley, CA 94720

kutschke commented:
Will do.  I just had 5 runs in a row with no repeat.  But I can do this while watching the hockey game.


From: David Brown ***@***.***>
Reply-To: Mu2e/Offline ***@***.***>
Date: Tuesday, June 22, 2021 at 9:45 PM
To: Mu2e/Offline ***@***.***>
Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
Subject: Re: [Mu2e/Offline] Production Fixes (#501)

Could you attach gdb when it hangs and find out what function it's in?

On Tue, Jun 22, 2021 at 7:37 PM Rob Kutschke ***@***.***>
wrote:

> I did the merge and build.
> I ran 100 events of ceSteps; it writes 58 events.
>
> Then I ran time mu2e -c Validation/ceDigi.fcl --trace
>
> This sometimes runs to completion and sometimes hangs in makeSD.
>
> It hangs maybe 1 time in 3 or 4. When it’s hanging it’s consuming cpu. So
> I presume infinite loop not IO wait.
>
> Looking to see if it stops at the same event or different ones/
>
>
>
> @reply.github.com>
> Date: Tuesday, June 22, 2021 at 9:07 PM
> To: Mu2e/Offline ***@***.***>
> Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
> Subject: Re: [Mu2e/Offline] Production Fixes (#501)
>
> I tested interactively on the same branch that has stalled in CI.
>
>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#issuecomment-866476011><https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866476011-253E&d=DwQFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Ot01EKJGyeUmAtuAfpvMjv0FcBT6ntkxMo-t8hTIO88&s=SAQxIlqtGymHrMjZ1dsWbjqc83HPwrkIuUm8KIFLjIM&e=>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH57YZHTWEYTUKV5EOO6TTUFCH7ANCNFSM47CRZ6HQ><https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABAH57YZHTWEYTUKV5EOO6TTUFCH7ANCNFSM47CRZ6HQ-253E&d=DwQFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Ot01EKJGyeUmAtuAfpvMjv0FcBT6ntkxMo-t8hTIO88&s=2KH7kvHYrtOtAHzEpZW2Y8U2FqN0K3J4ms0o4xyu1bQ&e=>
> .
>


--
David Nathan Brown ***@***.***
Office Phone (510) 486-7261 Fax 495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C) Berkeley, CA 94720


—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866478511&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Ot01EKJGyeUmAtuAfpvMjv0FcBT6ntkxMo-t8hTIO88&s=KEjNjYgZI0WrcVPJ3XEOn3KjYxF3LAGjD2B2Elb3koc&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZTQH5V7HL25MLRMN4TTUFDCVANCNFSM47CRZ6HQ&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Ot01EKJGyeUmAtuAfpvMjv0FcBT6ntkxMo-t8hTIO88&s=XnLM8gSVlPoLPTas9kQQcpTlqqZDBt3xnrBh1tUM0pc&e=>.

kutschke commented:
Adding Richie.

Bad news and good news.   I could not get it to hang in the debugger.  While I was rerunning it I built the head of master and started testing it.  It hung on every trial
At event 65 of the ceSteps output.

Here is the tail end of the art –trace output just before the hang, followed by the gdb traceback:


++++ finished event

++++ source event

++++ finished source event

++++ processing event: run: 1000 subRun: 0 event: 65 time: 0

++++++ processing path for event: DigitizePath

++++++ processing path for event: cprLowPSeedDeM_trigger

++++++ processing path for event: cprLowPSeedDeP_trigger

++++++ processing path for event: cprSeedDeM_trigger

++++++ processing path for event: cprSeedDeP_trigger

++++++ processing path for event: largeCDCount_trigger

++++++ processing path for event: largeSDCount_trigger

++++++ processing path for event: minimumbiasCDCount_trigger

++++++ processing path for event: minimumbiasSDCount_trigger

++++++ processing path for event: tprHelixCalibIPADeM_trigger

++++++ processing path for event: tprHelixIPADeM_trigger

++++++ processing path for event: tprLowPSeedDeM_trigger

++++++ processing path for event: tprLowPSeedDeP_trigger

++++++ processing path for event: tprSeedDeM_trigger

++++++ processing path for event: tprSeedDeP_trigger

++++++ processing path for event: unbiased_trigger

++++++++ module for event: EWMProducer

++++++++ finished for event: EWMProducer

++++++++ module for event: makeSD

^C

Program received signal SIGINT, Interrupt.

0x00007ffff5189241 in __ieee754_exp_avx () from /lib64/libm.so.6

(gdb) where

#0  0x00007ffff5189241 in __ieee754_exp_avx () from /lib64/libm.so.6

#1  0x00007ffff5153be3 in exp () from /lib64/libm.so.6

#2  0x00007fffe8726d9b in mu2e::StrawElectronics::linearResponse ***@***.***=0x237ca250, straw=..., ***@***.***=mu2e::StrawElectronics::thresh, time=<optimized out>, charge=0.010576232336461544,

    distance=814.85687255859375, ***@***.***=false) at Offline/TrackerConditions/src/StrawElectronics.cc:85

#3  0x00007fffd8c00e41 in mu2e::TrackerMC::StrawWaveform::sampleWaveform ***@***.***=0x7ffffffeefa0, strawele=..., ***@***.***=mu2e::StrawElectronics::thresh, time=<optimized out>)

    at Offline/TrackerMC/inc/StrawCluster.hh:50

#4  0x00007fffd8c00f8c in mu2e::TrackerMC::StrawWaveform::returnCrossing ***@***.***=0x7ffffffeefa0, strawele=..., ***@***.***=9.2821176052093506, wfx=...)

    at Offline/TrackerConditions/inc/StrawElectronics.hh:158

#5  0x00007fffd8c013bb in mu2e::TrackerMC::StrawWaveform::crossesThreshold ***@***.***=0x7ffffffeefa0, strawele=..., ***@***.***=9.2821176052093506, wfx=...)

    at Offline/TrackerMC/src/StrawWaveform.cc:36

#6  0x00007fffd8e381a1 in mu2e::TrackerMC::StrawDigisFromStrawGasSteps::findThresholdCrossings ***@***.***=0x1bac57f0, strawele=..., swfp=..., xings=...)

    at Offline/TrackerMC/src/StrawDigisFromStrawGasSteps_module.cc:746

#7  0x00007fffd8e3e5bf in mu2e::TrackerMC::StrawDigisFromStrawGasSteps::createDigis (this=0x1bac57f0, strawphys=..., strawele=..., tracker=..., straw=..., hsp=..., xtalk=..., digis=0x23aa34c0, digiadcs=0x2a4a1680,

    mcdigis=0x236e8450) at Offline/TrackerMC/src/StrawDigisFromStrawGasSteps_module.cc:530

#8  0x00007fffd8e3f2b2 in mu2e::TrackerMC::StrawDigisFromStrawGasSteps::produce (this=0x1bac57f0, event=...) at Offline/TrackerMC/src/StrawDigisFromStrawGasSteps_module.cc:487

#9  0x00007ffff4611d3a in art::EDProducer::produceWithFrame (this=0x1bac57f0, e=..., frame=...)

    at /scratch/workspace/critic-slf/BUILDTYPE/prof/QUAL/e20/label1/swarm/label2/SLF7/build/art/v3_09_01-buildFW/src/art/Framework/Core/EDProducer.cc:86

#10 0x00007ffff4691080 in art::detail::Producer::doEvent (this=0x1bac57f0, ep=..., mc=..., counts_run=..., counts_passed=...)

    at /scratch/workspace/critic-slf/BUILDTYPE/prof/QUAL/e20/label1/swarm/label2/SLF7/build/art/v3_09_01-buildFW/src/art/Framework/Core/detail/Producer.cc:115

#11 0x00007ffff3935c47 in art::Worker::runWorker (this=0x1bad8390, p=..., mc=...)

    at /scratch/workspace/critic-slf/BUILDTYPE/prof/QUAL/e20/label1/swarm/label2/SLF7/build/art/v3_09_01-buildFW/src/art/Framework/Principal/Worker.cc:483

#12 0x00007ffff3935ece in art::Worker::<lambda()>::operator() (__closure=<optimized out>)

    at /scratch/workspace/critic-slf/BUILDTYPE/prof/QUAL/e20/label1/swarm/label2/SLF7/build/art/v3_09_01-buildFW/src/art/Framework/Principal/Worker.cc:621


From: David Brown ***@***.***>
Reply-To: Mu2e/Offline ***@***.***>
Date: Tuesday, June 22, 2021 at 9:45 PM
To: Mu2e/Offline ***@***.***>
Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
Subject: Re: [Mu2e/Offline] Production Fixes (#501)

Could you attach gdb when it hangs and find out what function it's in?

On Tue, Jun 22, 2021 at 7:37 PM Rob Kutschke ***@***.***>
wrote:

> I did the merge and build.
> I ran 100 events of ceSteps; it writes 58 events.
>
> Then I ran time mu2e -c Validation/ceDigi.fcl --trace
>
> This sometimes runs to completion and sometimes hangs in makeSD.
>
> It hangs maybe 1 time in 3 or 4. When it’s hanging it’s consuming cpu. So
> I presume infinite loop not IO wait.
>
> Looking to see if it stops at the same event or different ones/
>
>
>
> @reply.github.com>
> Date: Tuesday, June 22, 2021 at 9:07 PM
> To: Mu2e/Offline ***@***.***>
> Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
> Subject: Re: [Mu2e/Offline] Production Fixes (#501)
>
> I tested interactively on the same branch that has stalled in CI.
>
>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Mu2e/Offline/pull/501#issuecomment-866476011><https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866476011-253E&d=DwQFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Ot01EKJGyeUmAtuAfpvMjv0FcBT6ntkxMo-t8hTIO88&s=SAQxIlqtGymHrMjZ1dsWbjqc83HPwrkIuUm8KIFLjIM&e=>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAH57YZHTWEYTUKV5EOO6TTUFCH7ANCNFSM47CRZ6HQ><https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABAH57YZHTWEYTUKV5EOO6TTUFCH7ANCNFSM47CRZ6HQ-253E&d=DwQFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Ot01EKJGyeUmAtuAfpvMjv0FcBT6ntkxMo-t8hTIO88&s=2KH7kvHYrtOtAHzEpZW2Y8U2FqN0K3J4ms0o4xyu1bQ&e=>
> .
>


--
David Nathan Brown ***@***.***
Office Phone (510) 486-7261 Fax 495-2957
Lawrence Berkeley National Lab
MS 50R5008 (50-6026C) Berkeley, CA 94720


—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_501-23issuecomment-2D866478511&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Ot01EKJGyeUmAtuAfpvMjv0FcBT6ntkxMo-t8hTIO88&s=KEjNjYgZI0WrcVPJ3XEOn3KjYxF3LAGjD2B2Elb3koc&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZTQH5V7HL25MLRMN4TTUFDCVANCNFSM47CRZ6HQ&d=DwMFaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=Ot01EKJGyeUmAtuAfpvMjv0FcBT6ntkxMo-t8hTIO88&s=XnLM8gSVlPoLPTas9kQQcpTlqqZDBt3xnrBh1tUM0pc&e=>.

kutschke commented:
We believe that the problem underlying the timeout in ceDIgi was introduced when we merged PR 491.  We have contacted the author of that PR, Richie, who says he knows what changes caused the problem and he will deal with it in the morning.  Since merging this PR will not introduce new problems I am merging it now.
